(CDAP-7823) Reported operational stats for CDAP

bdmogal opened the pull request:
- Includes an Operational Stats Framework change to add initialize and destroy methods to operational extensions.
- The CDAP master injector is available to operational stats extensions via the initialize method

Jira: [CDAP-7823](https://issues.cask.co/browse/CDAP-7823)
Build: http://builds.cask.co/browse/CDAP-RUT333

There is one more stat that we could possibly report for CDAP, but opening PR to get review started and also because this needs a small UI change.

Commit:
(CDAP-7823) Reported operational stats for CDAP

- Includes an Operational Stats Framework change to add initialize and destroy methods to operational extensions.
- The CDAP master injector is available to operational stats extensions via the initialize method
- Allow OperationalStats to throw exceptions other than IOException

albertshau commented on cdap-operational-stats-core/src/main/java/co/cask/cdap/operations/cdap/CDAPEntities.java:
```diff
@@ -0,0 +1,131 @@
+  @Override
+  public void collect() throws Exception {
+    reset();
+    NamespaceQueryAdmin nsQueryAdmin = injector.getInstance(NamespaceQueryAdmin.class);
```
is there a reason we need to get an instance for each collect call instead of at initialize?

bdmogal replied:
nope, can be moved to initialize. Will do.

albertshau reviewed (commented):
one question, rest lgtm

bdmogal commented:
Addressed comments. Build is also green. However, there are some pending UI changes because of this. @ajainarayanan @PGuimarais do you guys want to use this branch to push your fixes, or should I merge this?

bdmogal commented:
Green-- http://builds.cask.co/browse/CDAP-RUT333-9. Merging. UI changes will be in a separate PR.
